### PR TITLE
Support for SMaHT data model

### PR DESCRIPTION
**Pull request by alexander-veit**
This PR makes various changes to support the SMaHT data model. Changes to `tibanna_ffcommon` are minimal and should be compatible with the CGAP and 4DN layers. Most of the required changes are in in the `tibanna_tiger` module. Some functions had to be reimplemented in child classes in order to make them compatible with SMaHT.

Tests for `tibanna_tiger` are largly missing. I will add them in a later stage.

### CODE REVIEW

File: tibanna_ffcommon/portal_utils.py [hunk 1]
```diff
@@ -1260,6 +1260,18 @@ def update_input_extras(self):
                     self.update_patch_items(ip['uuid'], {'higlass_uid': higlass_uid})
             self.update_patch_items(ip['uuid'], {'extra_files': ip['extra_files']})
 
+    def get_portal_specific_item_name(self, item: str) -> str:
+        """This function gets the portal specific item names for a given item name. This is necessary because
+        the nameing of similar items can vary across portals, e.g. "quality_metric_generic" (CGAP) vs 
+        "quality_metric" (SMaHT)
```
Comment (willronchetti) [inline tibanna_ffcommon/portal_utils.py:1266]: Is this update needed for `pony` as well?

File: tibanna_ffcommon/vars.py [hunk 2]
```diff
@@ -64,10 +64,13 @@ def BUCKET_NAME(env, filetype):
     global _BUCKET_NAME_LOG
     global _BUCKET_NAME_CWL
 
+    processed_file_types = ['FileProcessed', 'OutputFile']
+    raw_file_types = ['FileFastq', 'FileReference', 'ReferenceFile' 'FileMicroscopy', 'FileSubmitted']
```
Comment (willronchetti) [inline tibanna_ffcommon/vars.py:68]: Are these not also `SubmittedFile`, `ReferenceFile` etc in the same form as `OutputFile` above? Maybe my branch is out of date...

**Review by willronchetti — APPROVED** [review]
Small comments nothing major

File: tibanna_ffcommon/portal_utils.py (hunk 1 above)
Comment (alexander-veit) [reply tibanna_ffcommon/portal_utils.py:1266]: In pony we never reach `update_generic_qc` which is the only function that is using the mapping. I think we can add it to pony, when we actually need it.

File: tibanna_ffcommon/vars.py (hunk 2 above)
Comment (alexander-veit) [reply tibanna_ffcommon/vars.py]: Good catch. I added `SubmittedFile`. We haven't done any testing with those, that's why I was missing it.

File: tibanna_smaht/tiger_utils.py [hunk 3]
```diff
@@ -1,124 +1,275 @@
...
 )
+from tibanna import create_logger
+
+logger = create_logger(__name__)
+
+
+class TigerInputFile(FFInputFile):
+    def get_file_format_from_uuid(self, uuid):
+        """returns parsed (cleaned) version of file format (e.g. 'bam')"""
+        file_format_uuid = parse_formatstr(self.get_metadata(uuid)["file_format"])
```
[the remark this answers is not in the document]
Comment (alexander-veit) [reply tibanna_smaht/tiger_utils.py:40]: `file_format` is a required property for the `file` type. So, I think this is safe.

**Pull request merged**